File: benchmark/python/substring_search.py
import time

text = ""
for i in range(1000):
    text = text + "abcdefghij"
pattern = "defg"
text_len = len(text)
pat_len = len(pattern)
count = 0
start = time.time() * 1000
for t in range(text_len - pat_len + 1):
    match = True
    for p in range(pat_len):
        if text[t + p] != pattern[p]:
            match = False
            break
    if match:
        count = count + 1
end = time.time() * 1000
print(f"Count: {count}")
print(f"Time: {end - start:.0f}ms")

File: benchmark/lua/substring_search.lua
local text = ""
for i = 1, 1000 do
    text = text .. "abcdefghij"
end
local pattern = "defg"
local text_len = #text
local pat_len = #pattern
local count = 0
local start = os.clock()
for t = 1, text_len - pat_len + 1 do
    local match = true
    for p = 1, pat_len do
        if string.sub(text, t + p - 1, t + p - 1) ~= string.sub(pattern, p, p) then
            match = false
            break
        end
    end
    if match then
        count = count + 1
    end
end
local finish = os.clock()
print("Count: " .. count)
print(string.format("Time: %.0fms", (finish - start) * 1000))
